### PR TITLE
[enterprise-metrics] Upgrade to GEM 1.6.1

### PR DESCRIPTION
**Pull request by Logiraptor**
Signed-off-by: Patrick Oyarzun <patrick.oyarzun@grafana.com>

### CODE REVIEW

**Comment by chaudum** [comment]
Hey @Logiraptor could you also add an entry in a new `1.7.0` version section to the changelog?

**Review by chaudum — CHANGES_REQUESTED** [review]
Version upgrade looks good.

However, I upgraded from 1.6.0 to 1.7.0 locally with a default configuration. The `query-frontend` service panics and the pod crash-loops:

```console
$ k logs --namespace helm-metrics haudum-enterprise-metrics-query-frontend-597c7d59-cgjr6
level=warn ts=2021-11-24T08:25:38.569739286Z caller=experimental.go:24 msg="experimental feature in use" feature=tenant-federation
level=info ts=2021-11-24T08:25:38.569825896Z caller=main.go:211 msg="starting Grafana Enterprise Metrics" version="(version=v1.6.1, branch=HEAD, revision=6a547d1cd)"
level=info ts=2021-11-24T08:25:38.569977684Z caller=filesystem.go:37 msg="excluding mount points" mounts=^/(dev|proc|sys|var/lib/docker/.+)($|/)
level=info ts=2021-11-24T08:25:38.570004349Z caller=filesystem.go:40 msg="excluding filesystem types" types=^(autofs|binfmt_misc|bpf|cgroup2?|configfs|debugfs|devpts|devtmpfs|fusectl|hugetlbfs|iso9660|mqueue|nsfs|overlay|proc|procfs|pstore|rpc_pipefs|securityfs|selinuxfs|squashfs|sysfs|tracefs)$
level=info ts=2021-11-24T08:25:38.5708388Z caller=quota.go:297 msg="registered quota collector"
level=warn ts=2021-11-24T08:25:38.571277722Z caller=manager.go:173 msg="unable to sync local license to admin storage backend" error="license 4819 with cluster name metrics-enterprise-test-fixture != haudum"
level=warn ts=2021-11-24T08:25:38.573251162Z caller=manager.go:218 msg="invalid license present" name=11324 error="license 4819 with cluster name metrics-enterprise-test-fixture != haudum"
level=warn ts=2021-11-24T08:25:38.573271858Z caller=added_modules.go:715 msg="unable to load config extensions due to no valid license being found"
level=info ts=2021-11-24T08:25:38.573393516Z caller=server.go:269 http=[::]:8080 grpc=[::]:9095 msg="server listening on addresses"
level=error ts=2021-11-24T08:25:38.573824792Z caller=added_modules.go:173 msg="error initializing the TSDB meta exporter." err="failed to create bucket client: no s3 endpoint in config file"
level=error ts=2021-11-24T08:25:38.575330059Z caller=query_logger.go:108 msg="Failed to create directory for logging active queries"
level=error ts=2021-11-24T08:25:38.575348342Z caller=query_logger.go:86 msg="Error opening query log file" file=/data/enterprise-metrics/querier/queries.active err="open /data/enterprise-metrics/querier/queries.active: no such file or directory"
panic: Unable to create mmap-ed active query log

goroutine 1 [running]:
github.com/prometheus/prometheus/promql.NewActiveQueryTracker(0xc000a15720, 0x20, 0x14, 0x306c7a0, 0xc000504050, 0x10)
	/backend-enterprise/vendor/github.com/prometheus/prometheus/promql/query_logger.go:116 +0x4e8
github.com/grafana/mimir/pkg/querier/engine.createActiveQueryTracker(...)
	/backend-enterprise/vendor/github.com/grafana/mimir/pkg/querier/engine/config.go:76
github.com/grafana/mimir/pkg/querier/engine.NewPromQLEngineOptions(0x14, 0x1bf08eb000, 0x2faf080, 0x0, 0xdf8475800, 0xc000a15720, 0x20, 0x45d964b800, 0x306c7a0, 0xc000504050, ...)
	/backend-enterprise/vendor/github.com/grafana/mimir/pkg/querier/engine/config.go:61 +0x230
github.com/grafana/mimir/pkg/mimir.(*Mimir).initQueryFrontendTripperware(0xc00095e000, 0x309d820, 0xc000711590, 0xc00095e000, 0xc000a78410)
	/backend-enterprise/vendor/github.com/grafana/mimir/pkg/mimir/modules.go:520 +0x105
github.com/grafana/dskit/modules.(*Manager).initModule(0xc00000d848, 0x7ffd00c48605, 0xe, 0xc0009854a8, 0xc000692f60, 0x0, 0x0)
	/backend-enterprise/vendor/github.com/grafana/dskit/modules/modules.go:106 +0x2ea
github.com/grafana/dskit/modules.(*Manager).InitModuleServices(0xc00000d848, 0xc000aa00a0, 0x1, 0x1, 0xc000a78230, 0xc000a7fb01, 0x170)
	/backend-enterprise/vendor/github.com/grafana/dskit/modules/modules.go:78 +0x105
github.com/grafana/mimir/pkg/mimir.(*Mimir).Run(0xc00095e000, 0xc0006a6f00, 0x4)
	/backend-enterprise/vendor/github.com/grafana/mimir/pkg/mimir/mimir.go:410 +0x2a5
github.com/grafana/backend-enterprise/pkg/enterprise/mimir/init.(*MimirEnterprise).Run(...)
	/backend-enterprise/pkg/enterprise/mimir/init/mimir.go:147
main.main()
	/backend-enterprise/cmd/enterprise-metrics/main.go:212 +0x1048
```

Could you investigate?

**Comment by Logiraptor** [comment]
@chaudum Sorry about that! The query_frontend now makes use of the active query tracker and therefore needs a writable data volume. I've added that in and it should be working now.

File: charts/enterprise-metrics/templates/query-frontend-dep.yaml [hunk 1]
```diff
@@ -80,6 +80,9 @@ spec:
               mountPath: /license
             - name: config
               mountPath: /etc/enterprise-metrics
+            - name: storage
+              mountPath: /data
+              subPath: {{ .Values.query_frontend.persistence.subPath }}
```
Comment (chaudum) [inline charts/enterprise-metrics/templates/query-frontend-dep.yaml:85]: I think it makes sense to only conditionally set the subPath:

```suggestion
              {{- if .Values.query_frontend.persistence.subPath }}
              subPath: {{ .Values.query_frontend.persistence.subPath }}
              {{- end }}
```

**Review by chaudum — COMMENTED** [review]
Otherwise LGTM

**Review by chaudum — APPROVED** [review]
lgtm

**Pull request merged**